### PR TITLE
enable passes via env vars

### PR DESCRIPTION
**Pull request by bani-intelaipg**
Default pass enable/disable settings are applied for all unspecified options via command line env-var.
For example, NGRAPH_PASS_ENABLES="TransposeSinking:0" will set TransposeSinking off, and use the deafult for ConstantFolding and others. Deafult values are specified in TranslateGraph()

### CODE REVIEW

**Review by adk9 — CHANGES_REQUESTED** [review]

File: ngraph_bridge/ngraph_builder.cc [hunk 1]
```diff
@@ -3347,7 +3347,8 @@ Status Builder::TranslateGraph(
   // Apply additional passes on the nGraph function here.
   //
   ngraph::pass::Manager passes;
-  // passes.register_pass<ngraph::pass::ConstantFolding>();
+  if (PassEnabled(NGraphPassType::ConstFold))
+    passes.register_pass<ngraph::pass::ConstantFolding>();
   passes.register_pass<pass::TransposeFolding>();
   passes.register_pass<pass::TransposeSinking>();
```
[the remark this answers is not in the document]
Comment (bani-intelaipg) [reply ngraph_bridge/ngraph_builder.cc:3353]: Are these are the only 3 as of now?

File: ngraph_bridge/ngraph_utils.cc [hunk 2]
```diff
@@ -32,6 +32,8 @@
 #include "ngraph_bridge/ngraph_utils.h"
```
[the remark this answers is not in the document]
Comment (adk9) [reply ngraph_bridge/ngraph_utils.cc:32]: `ngraph_utils.*` is a kitchen sink already. I think that it's preferable to have helper methods such as these in the proper ngraph_bridge namespace instead of putting them all in utils.

File: ngraph_bridge/ngraph_utils.cc [hunk 3]
```diff
@@ -586,6 +589,30 @@ void ClearAttribute(Graph* g,
   }
 }
 
+static ngraph::pass::PassConfig* _p_pass_config = nullptr;
```
Comment (adk9) [inline ngraph_bridge/ngraph_utils.cc:592]: Singleton config is not required. You should check for enabled passes and attributes everytime `TranslateGraph()` is called.

File: ngraph_bridge/ngraph_utils.cc [hunk 4]
```diff
@@ -586,6 +589,30 @@ void ClearAttribute(Graph* g,
...
+
+static void InitPassConfig() {
+  // use a singleton config object
+  if (_p_pass_config) return;
+  _p_pass_config = new ngraph::pass::PassConfig();
+  // Set the defaults, unless already specified via env var
+  if (ngraph::getenv_string("NGRAPH_PASS_ENABLES").empty()) {
```
Comment (adk9) [inline ngraph_bridge/ngraph_utils.cc:599]: This is error-prone. Use `PassConfig` to find out whether a pass was enabled or not.

**Review by adk9 — CHANGES_REQUESTED** [review]

File: ngraph_bridge/ngraph_utils.cc (hunk 3 above)
Comment (bani-intelaipg) [reply ngraph_bridge/ngraph_utils.cc:592]: We can/may call PassEnabled() many times as needed. The singleton is to just read the env-var once at the very beginning. If we had a system-init kind of func, we could init it there too.

**Review by adk9 — APPROVED** [review]

File: ngraph_bridge/ngraph_builder.cc [hunk 5]
```diff
@@ -3257,11 +3258,24 @@ Status Builder::TranslateGraph(
...
-  passes.run_passes(ng_function);
+  {
+    ngraph::pass::Manager passes;
+    ngraph::pass::PassConfig pass_config;
+    // set/honor the defaults, unless specified via env var
+    auto set_default = [&pass_config](std::string pass, bool enable) {
+      auto enables_map = pass_config.get_enables();
+      if (enables_map.find(pass) == enables_map.end())
+        pass_config.set_pass_enable(pass, enable);
+    };
```
Comment (adk9) [inline ngraph_bridge/ngraph_builder.cc:3269]: Minor comment: This is fine, but if you change it to `get_enabled(std::string, bool default)` then there's only one change required to enable/disable a pass.

File: test/test_ngraph_exec.cpp [hunk 6]
```diff
@@ -36,6 +41,14 @@ namespace tensorflow {
...
 namespace testing {
 
+string THISDIR =
+#if defined(WIN32) || defined(_WIN32) || \
+    defined(__WIN32) && !defined(__CYGWIN__)
+    std::regex_replace(__FILE__, std::regex("^(.*)\\[^\\]+$"), "$1\\");
+#else
+    std::regex_replace(__FILE__, std::regex("^(.*)/[^/]+$"), "$1/");
+#endif
+
```
Comment (adk9) [inline test/test_ngraph_exec.cpp:51]: Why is this required?

Comment (bani-intelaipg) [reply test/test_ngraph_exec.cpp:51]: I removed path dependency of running gtest from some other folder e.g. source root. Now, it will always find the associated pbtxt files in the same dir as the test script.

Comment (adk9) [reply test/test_ngraph_exec.cpp:53]: This creates a dependency on the source. If you want to not use a path relative to CWD, please use a path relative to the test executable.

**Review by adk9 — CHANGES_REQUESTED** [review]

File: test/test_ngraph_exec.cpp (hunk 6 above)
Comment (bani-intelaipg) [reply test/test_ngraph_exec.cpp:53]: I will revert to older approach of copying pbtxt files to build output.

**Review by adk9 — APPROVED** [review]

**Review by kanvi-nervana — APPROVED** [review]
